src/blocks/image-comparison/components/Edit.js

```diff
@@ -46,10 +45,46 @@ const Edit = ({ attributes, setAttributes, clientId }) => {
     customCaptionTextColour,
     captionBackgroundColour,
     customCaptionBackgroundColour,
-    containerHeight,
-    containerWidth,
   } = attributes;
 
+  /**
+   * Get the container's size. If this has not been set by
+   * the user then we overwrite the default size of the block
+   * with the theme's defined contentSize, if it exists.
+   *
+   * @returns {
+   *  width: string
+   *  height: string
+   * } The size containing the height and width of the block's container.
+   */
+  const getContainerSize = () => {
+    if (attributes.align) {
+      return {
+        containerWidth: 'auto',
+        containerHeight: 'auto',
+      };
+    }
+
+    let containerHeight = '500px';
+    if (attributes.containerHeight) {
+      containerHeight = attributes.containerHeight;
+    }
+
+    let containerWidth = '500px';
+    if (attributes.containerWidth) {
+      containerWidth = attributes.containerWidth;
+    } else if (contentWidth) {
+      containerWidth = contentWidth;
+    }
+
+    return {
+      containerHeight,
+      containerWidth,
+    };
+  };
+
+  const { containerHeight, containerWidth } = getContainerSize();
```

Comment:
I don't think this function will be needed? If the `containerHeight` and `containerWidth` attributes aren't set then the block just needs to respond to it's parent element, the nullish coalescing operator (`??`) might hopefully be enough to handle this?

I'm not sure where the `500px` default width and height values have come from as currently this will lead to a visual discrepancy between editor and frontend, steps to reproduce this;

- Add Image Comparison block
- Add 2 images
- Save

The block in editor will end up with the height of `500px`, if you then visit the frontend you'll see that the block is displaying as expected (height scaling with the width), are the `500px` values required?

Reply:
Hey @PaulREnglish thanks for the explanation, we should try and avoid setting any width or height pixel values until the user specifically sets them and let the image respond to it's container by default.

I'm not sure of the reason they were set to default to `500px` originally as well, I may be overlooking something here but please let me know if I can assist with anything on this 👍

Reply:
Will do thanks!